ethcore/wasm/src/lib.rs

```diff
@@ -79,8 +79,8 @@ impl vm::Vm for WasmInterpreter {
 			&wasmi::ImportsBuilder::new().with_resolver("env", &instantiation_resolover)
 		).map_err(Error)?;
 
-		let adjusted_gas = params.gas * U256::from(ext.schedule().wasm.opcodes_div) /
-			U256::from(ext.schedule().wasm.opcodes_mul);
+		let adjusted_gas = params.gas * U256::from(ext.schedule().wasm().opcodes_div) /
```

Comment:
Each call to `ext.schedule()` is virtual, can we cache it in a variable somewhere at the top?

Reply:
I don't think so, it's mutably borrows self all the way down the end of the function few times

I think it will be also optimized / devirtualized anyway